tests/unit/test_utils.py

```diff
@@ -41,6 +46,46 @@ def test_get_dataset_scenario():
         get_dataset_scenario(data, 2)
 
 
+@pytest.fixture
+def batch_data() -> Dict[str, np.ndarray]:
+    line = initialize_array("update", "line", (3, 2))
+    line["id"] = [[5, 6], [6, 7], [7, 5]]
+    line["from_status"] = [[1, 1], [1, 1], [1, 1]]
+
+    # Add batch for asym_load, which has 2-D array for p_specified
+    asym_load = initialize_array("update", "asym_load", (3, 2))
+    asym_load["id"] = [[9, 10], [9, 10], [9, 10]]
+
+    return {"line": line, "asym_load": asym_load}
+
+
+def test_get_dataset_batch_size(batch_data):
+    assert get_dataset_batch_size(batch_data) == 3
+
+
+def test_get_dataset_batch_size_sparse():
+    batch_size = 3
+    data = {
+        "node": {
+            "data": np.zeros(shape=3, dtype=power_grid_meta_data["input"]["node"]),
+            "indptr": np.array([0, 2, 3, 3]),
+        },
+        "sym_load": {
+            "data": np.zeros(shape=2, dtype=power_grid_meta_data["input"]["sym_load"]),
+            "indptr": np.array([0, 0, 1, 2]),
+        },
+        "asym_load": {
+            "data": np.zeros(shape=4, dtype=power_grid_meta_data["input"]["asym_load"]),
+            "indptr": np.array([0, 2, 3, 4]),
+        },
+        "link": np.zeros(shape=(batch_size, 4), dtype=power_grid_meta_data["input"]["link"]),
+    }
+
+    dataset = CConstDataset(data, ["input"])
+
+    assert get_dataset_batch_size(dataset) == 3
```

Comment:
please also add a bad weather test case that checks that the correct behavior happens when the batch size is inconsistent. Most importantly, the error should be consistent with the get_scenario function.

If i'm correct, then that function raises a `KeyError`, meaning this function should raise a `ValueError` (due to inconsistent batch size)

raising tests can be tested using

```py
with pytest.raises(ValueError, msg):
    call()
```
if you look for `pytest.raises` you'll find plenty of examples

Reply:
I am not sure by this way of constructing sparse dataset, an `inconsistent` complaint would be given. I assume you are referring to construct a inconsistent `Dict[str, np.ndarray]`  instead of a sparse dataset?

Reply:
well, both actually:

the input for `get_dataset_scenario` has type `dataset: Dict[str, ndarray | Dict[str, ndarray]]` (see also https://power-grid-model.readthedocs.io/en/stable/api_reference/python-api-reference.html#power_grid_model.utils.get_dataset_scenario ).

The different types are given in https://github.com/PowerGridModel/power-grid-model/blob/main/src/power_grid_model/data_types.py (it may be good to add that file to the docs in https://github.com/PowerGridModel/power-grid-model/blob/main/docs/api_reference/python-api-reference.md )

In particular, there are 2 batch array types that both may have consistent and inconsistent batch size. E.g. (modulo correct data types):

* Fully dense inconsistent update dataset: `{"sym_power_sensor": np.array([[]]), "asym_power_sensor": np.array([[], []])}`
  * `sym_power_sensor` has batch size 1 while `asym_power_sensor` has batch size 2
* Fully sparse inconsistent update dataset: `{"sym_power_sensor": {"indptr": np.array([0, 0]), "data": np.array([])}, "asym_power_sensor": {"indptr": np.array([0, 0, 0]), "data": np.array([])}}`
  * idem
* Mixed inconsistent update dataset: `{"sym_power_sensor": np.array([[]]),  "asym_power_sensor": {"indptr": np.array([0, 0, 0]), "data": np.array([])}}}`

for easier creation, you can also use the `initialize_array` function. the `shape` argument may help there